### ISSUE TITLE
Uncaught Error: toString failed

### ISSUE THREAD

**Issue by hank-smatoos** [v3ss0n/goto#86]
[Enter steps to reproduce below:]
1. int the following PHP source code, I selected 'getContentSearch'.
   `$data['query']     = $this->boardmodel->getContentSearch($getDate);`
2. execute 'go to declaration' through menu

**Atom Version**: 1.6.0
**System**: Mac OS X 10.11.3
**Thrown From**: [goto](https://github.com/v3ss0n/goto) package, v1.7.1
### Stack Trace

Uncaught Error: toString failed

```
At buffer.js:378

Error: toString failed
    at Buffer.toString (buffer.js:378:11)
    at Object.fs.readFileSync (fs.js:461:33)
    at Object.fs.readFileSync (ATOM_SHELL_ASAR.js:385:29)
    at SymbolIndex.module.exports.SymbolIndex.processFile (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:182:15)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:173:12)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.rebuild (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:117:8)
    at SymbolIndex.module.exports.SymbolIndex.update (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:109:8)
    at SymbolIndex.module.exports.SymbolIndex.gotoDeclaration (/Users/hank/.atom/packages/goto/lib/symbol-index.coffee:129:6)
    at Object.module.exports.gotoDeclaration (/Users/hank/.atom/packages/goto/lib/index.coffee:36:22)
    at atom-workspace.atom.commands.add.goto:declaration (/Users/hank/.atom/packages/goto/lib/index.coffee:22:31)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (/Applications/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:1013:28)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
```
### Commands

```
     -6:05.8.0 core:copy (atom-text-editor.editor.is-focused)
     -5:52.1.0 editor:select-to-end-of-line (atom-text-editor.editor.is-focused)
     -1:43.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:20.9.0 core:close (atom-text-editor.editor.is-focused)
     -1:14.1.0 goto:declaration (atom-text-editor.editor.is-focused)
     -0:59.7.0 symbols-view:go-to-declaration (atom-text-editor.editor.is-focused)
     -0:53.7.0 goto:declaration (atom-text-editor.editor.is-focused)
     -0:33.8.0 core:select-to-bottom (atom-text-editor.editor.is-focused)
     -0:32.9.0 core:move-up (atom-text-editor.editor.is-focused)
     -0:32.1.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:29.3.0 goto:declaration (atom-text-editor.editor.is-focused)
     -0:27.7.0 editor:move-line-down (atom-text-editor.editor.is-focused)
     -0:26.8.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:26.8.0 core:cancel (atom-text-editor.editor.is-focused)
     -0:25.4.0 core:undo (atom-text-editor.editor.is-focused)
  2x -0:19.5.0 goto:declaration (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {}
}
```
### Installed Packages

``` coffee
# User
atom-sync, v0.6.0
goto, v1.7.1
remote-edit, v1.8.24

# Dev
No dev packages
```

**Comment by CaffeineSqurr** [comment]
[Enter steps to reproduce below:]

1. invalidate the goto index
2. rebuild the goto index

**Atom Version**: 1.6.0
**System**: Ubuntu 14.04.4
**Thrown From**: [goto](https://github.com/v3ss0n/goto) package, v1.7.1


### Stack Trace

Uncaught Error: toString failed

```
At buffer.js:378

Error: toString failed
    at Buffer.toString (buffer.js:378:11)
    at Object.fs.readFileSync (fs.js:461:33)
    at Object.fs.readFileSync (ATOM_SHELL_ASAR.js:385:29)
    at SymbolIndex.module.exports.SymbolIndex.processFile (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:182:15)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:173:12)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.rebuild (/home/cruise/.atom/packages/goto/lib/symbol-index.coffee:117:8)
    at atom-workspace.atom.commands.add.goto:rebuild-index (/home/cruise/.atom/packages/goto/lib/index.coffee:23:39)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at CommandPaletteView.module.exports.CommandPaletteView.confirmed (/usr/share/atom/resources/app.asar/node_modules/command-palette/lib/command-palette-view.js:183:32)
    at CommandPaletteView.module.exports.SelectListView.confirmSelection (/usr/share/atom/resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:338:21)
    at space-pen-div.atom.commands.add.core:confirm (/usr/share/atom/resources/app.asar/node_modules/atom-space-pen-views/lib/select-list-view.js:109:19)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:570:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:378:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -2:51.7.0 command-palette:toggle (div.panels-item)
     -2:47.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:47.4.0 goto:rebuild-index (div.panels-item)
     -1:05.5.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -1:04.8.0 core:move-down (atom-text-editor.editor.is-focused)
     -1:01.2.0 command-palette:toggle (input#core.followSymlinks.)
  2x -0:59.5.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:58 core:select-all (atom-text-editor.editor.mini.is-focused)
     -0:56.1.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:55.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:55.9.0 goto:invalidate-index (input#core.followSymlinks.)
     -0:54.8.0 command-palette:toggle (input#core.followSymlinks.)
 39x -0:51.6.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:48 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:45.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:45.4.0 goto:rebuild-index (input#core.followSymlinks.)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "destroyEmptyPanes": false,
    "followSymlinks": false,
    "ignoredNames": [
      ".*",
      "build",
      "devel",
      "logs"
    ],
    "packagesWithKeymapsDisabled": [
      "fuzzy-finder",
      "color-picker"
    ]
  }
}
```

### Installed Packages

```coffee
# User
clang-format, v1.25.0
color-picker, v2.2.2
file-icons, v1.6.20
goto, v1.7.1
highlight-line, v0.11.1
highlight-selected, v0.11.2
linter, v1.11.4
pigments, v0.31.1
sync-settings, v0.7.2
terminal-plus, v0.14.5

# Dev
No dev packages
```

**Comment by acusti** [comment]
Just published 1.8.0, which I’m hopeful resolves this issue. Please update if that isn’t the case.
